src/test/operators/delete_test.cpp

```diff
@@ -165,4 +167,68 @@ TEST_F(OperatorsDeleteTest, UpdateAfterDeleteFails) {
   t2_context->rollback();
 }
 
+TEST_F(OperatorsDeleteTest, DeleteOwnInsert) {
+  for (const auto value : {456.7, 457.7}) {
+    auto context = TransactionManager::get().new_transaction_context();
+
+    auto values_to_insert = load_table("src/test/tables/int_float3.tbl", Chunk::MAX_SIZE);
+    auto table_name_for_insert = "bla";
+    StorageManager::get().add_table(table_name_for_insert, values_to_insert);
+    auto insert_get_table = std::make_shared<GetTable>(table_name_for_insert);
+    insert_get_table->execute();
+
+    auto insert = std::make_shared<Insert>(_table_name, insert_get_table);
+    insert->set_transaction_context(context);
+    insert->execute();
+
+    auto validate1 = std::make_shared<Validate>(_gt);
+    validate1->set_transaction_context(context);
+    validate1->execute();
+
+    auto table_scan1 = std::make_shared<TableScan>(validate1, ColumnID{1}, PredicateCondition::Equals, value);
+    table_scan1->execute();
+    EXPECT_EQ(table_scan1->get_output()->row_count(), 2);
+
+    auto delete_op = std::make_shared<Delete>(_table_name, table_scan1);
+    delete_op->set_transaction_context(context);
+    delete_op->execute();
+
+    auto gt = std::make_shared<GetTable>(_table_name);
+    gt->execute();
+
+    auto validate2 = std::make_shared<Validate>(_gt);
+    validate2->set_transaction_context(context);
+    validate2->execute();
+
+    auto table_scan2 = std::make_shared<TableScan>(validate2, ColumnID{1}, PredicateCondition::Equals, value);
+    table_scan2->execute();
+    EXPECT_EQ(table_scan2->get_output()->row_count(), 0);
+
+    if (value == 456.7) {
```

Comment:
I get that you're trying to compress the code a bit here, but I think testing two different scenarios with a loop and an `if == MAGIC_FLOAT_NUMBER` is stretching it a bit. 
I'd prefer two test cases, I guess. 

Also, below you're comparing the result of both scenarios (commit and rollback) to the same result table (int_float_deleted.tbl). Imo the test should yield different values in each scenario, otherwise we can't be sure the rollback-path didn't just trigger the commit code erroneously?

Reply:
I don't have anything against testing both Insert and Delete in one test case, especially when there was a bug for this combination.

What really takes some effort to decipher is what this test is doing with a loop over two values and the rollback in one iteration and the commit in the other. Maybe a comment at the beginning of the test which explains the transactions conducted here would help. Along the lines of: "The transaction T consists of an Insert into a table and a subsequent selective Delete from this table. T is executed twice. The first execution is rolled back, the second is committed. This tests that deleting rows that were inserted in the same transaction works."

Reply:
better?